src/features/dialog-manager/reducer.js

```diff
@@ -165,10 +165,9 @@ const dialogManagerReducer = (state = initialState, { type, payload }) => {
             return {
                 ...state,
                 character: {
-                    ...character,
-                    characterName: payload.characterName,
-                    characterRace: payload.characterRace,
-                    characterClass: payload.characterClass,
+                    characterName: payload.characterName || characterName,
+                    characterRace: payload.characterRace || characterRace,
+                    characterClass: payload.characterClass || characterClass,
```

Comment:
What's the reasoning behind putting the '|| ' here? As far as I'm concerned, if the is undefined, null, or something else 'falsey' then that should be a bug in our logic. I'm happy to hear your thoughts though, because if there's a good reason we should consider adding this to the majority of the reducers.

Good catch on the characterName though.

Reply:
The reasoning is to force the `character[Name|Race|Class]` variables to be read, which removes the warnings. As you've pointed out the second value is only used if there's something 'wrong' with the first value - something that shouldn't happen given our logic. Although I'm tempted to suggest that a stray runtime error could produce something 'falsey' for the first value, I haven't seen this in testing.

Quite simply, it was the most straightforward way I saw to remove the warnings.

Reply:
Why not just remove line 69?

Reply:
You could have removed the warnings by just removing line 69.

Reply:
Upon reviewing my solution, I agree.